src/baseline/cannon/codegen.rs

```diff
@@ -0,0 +1,144 @@
+    pub lbl_return: Option<Label>,
+
+    // length of active_finallys in last loop
+    // default: 0
+    // break/continue need to emit finally blocks up to the last loop
+    // see tests/finally/break-while.dora
+    pub active_loop: Option<usize>,
+
+    // upper length of active_finallys in emitting finally-blocks for break/continue
+    // default: active_finallys.len()
+    // break/continue needs to execute finally-blocks in loop, return in these blocks
+    // would dump all active_finally-entries from the loop but we need an upper bound.
+    // see emit_finallys_within_loop and tests/finally/continue-return.dora
+    pub active_upper: Option<usize>,
+
+    pub cls_type_params: &'a TypeParams,
+    pub fct_type_params: &'a TypeParams,
+}
+
+impl<'a, 'ast> CannonCodeGen<'a, 'ast>
+where
+    'ast: 'a,
+{
+    fn emit_prolog(&mut self, bytecode: &BytecodeFunction) {
+        self.asm.prolog(bytecode.stacksize());
+        self.asm.emit_comment(Comment::Lit("prolog end"));
+        self.asm.emit_comment(Comment::Newline);
+    }
+
+    fn emit_epilog(&mut self, bytecode: &BytecodeFunction) {
+        self.asm.emit_comment(Comment::Newline);
+        self.asm.emit_comment(Comment::Lit("epilog"));
+
+        let polling_page = self.vm.polling_page.addr();
+        self.asm
+            .epilog_with_polling(bytecode.stacksize(), polling_page);
+    }
+
+    fn emit_const_bool(&mut self, bytecode: &BytecodeFunction, dest: &Register, bool_const: bool) {
+        let Register(index) = *dest;
+        let bytecode_type = bytecode.registers().get(index).expect("register not found");
+        let offset = bytecode.offset().get(index).expect("offset not found");
+
+        if bool_const {
+            self.asm.load_true(REG_RESULT);
+        } else {
+            self.asm.load_false(REG_RESULT);
+        }
+        self.asm
+            .store_mem(bytecode_type.mode(), Mem::Local(*offset), REG_RESULT.into());
+    }
+}
+
+impl<'a, 'ast> CodeGen<'ast> for CannonCodeGen<'a, 'ast> {
+    fn generate(mut self) -> JitBaselineFct {
+        let bytecode = generate_fct(
+            self.vm,
+            self.fct,
+            self.src,
+            self.cls_type_params,
+            self.fct_type_params,
+        );
+
+        if should_emit_bytecode(self.vm, self.fct) {
+            bytecode.dump();
+        }
+
+        if should_emit_debug(self.vm, self.fct) {
+            self.asm.debug();
+        }
+
+        self.emit_prolog(&bytecode);
+        // self.store_register_params_on_stack();
+        for btcode in bytecode.code() {
+            match btcode {
+                Bytecode::ConstTrue(dest) => self.emit_const_bool(&bytecode, &dest, true),
+                Bytecode::ConstFalse(dest) => self.emit_const_bool(&bytecode, &dest, true),
+                Bytecode::RetVoid => {}
+                _ => panic!("bytecode not implemented"),
+            }
+        }
+
+        let always_returns = self.src.always_returns;
+
+        if !always_returns {
```

Comment:
We shouldn't use always_returns in the new baseline compiler. The generated bytecode should always contain a `ret` at the end of the function.